nmigen_boards/arty_a7.py

```diff
@@ -0,0 +1,252 @@
+import os
+import subprocess
+
+from nmigen.build import *
+from nmigen.vendor.xilinx_7series import *
+from .dev import *
+
+
+__all__ = ["ArtyA7Platform"]
+
+
+class ArtyA7Platform(Xilinx7SeriesPlatform):
+    device     = "xc7a35ti"
+    package    = "csg324"
+    speedgrade = "1L"
+    resources  = [
+        Resource("clk100", 0, Pins("E3", dir="i"), Clock(100e6), Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("user_led", 0, Pins("H5",  dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_led", 1, Pins("J5",  dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_led", 2, Pins("T9",  dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_led", 3, Pins("T10", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("rgb_led", 0,
+            Subsignal("r", Pins("G6", dir="o")),
+            Subsignal("g", Pins("F6", dir="o")),
+            Subsignal("b", Pins("E1", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("rgb_led", 1,
+            Subsignal("r", Pins("G3", dir="o")),
+            Subsignal("g", Pins("J4", dir="o")),
+            Subsignal("b", Pins("G4", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("rgb_led", 2,
+            Subsignal("r", Pins("J3", dir="o")),
+            Subsignal("g", Pins("J2", dir="o")),
+            Subsignal("b", Pins("H4", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("rgb_led", 3,
+            Subsignal("r", Pins("K1", dir="o")),
+            Subsignal("g", Pins("H6", dir="o")),
+            Subsignal("b", Pins("K2", dir="o")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("user_sw", 0, Pins("A8" , dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_sw", 1, Pins("C11", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_sw", 2, Pins("C10", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_sw", 3, Pins("A10", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("user_btn", 0, Pins("D9", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_btn", 1, Pins("C9", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_btn", 2, Pins("B9", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("user_btn", 3, Pins("B8", dir="i"), Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("serial", 0,
+            Subsignal("tx", Pins("D10", dir="o")),
+            Subsignal("rx", Pins("A9",  dir="i")),
+            Attrs(IOSTANDARD="LVCMOS33")
+        ),
+
+        Resource("cpu_reset", 0, Pins("C2", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+
+        Resource("spi", 0,
```

Comment:
Would you be willing to add `SPIResource` like `SPIFlashResources`?

Reply:
Sure, I will do it in a later commit.